tests/unit/multi.tcl

```diff
@@ -34,12 +34,10 @@ start_server {tags {"multi"}} {
     } {QUEUED OK {a b c}}
 
     test {Nested MULTI are not allowed} {
-        set err {}
         r multi
-        catch {[r multi]} err
-        r exec
-        set _ $err
-    } {*ERR MULTI*}
```

Comment:
From looking at the tests I'm concerned this might be a breaking change, but people shouldn't rely on this errors.

Reply:
Honestly, I think we need to mark this as a breaking change. Someone could, wrongly, be executing:
```
MULTI
WATCH FOO
SET FOO BAR
EXEC
```

Which will error, but will work. With this change it won't work. We are allowed to make these types of changes, but I think we should be transparent and document them.

Reply:
yean, agree, lets' mark it with breaking changes, and document them you means we should put this change in the doc somewhere?

Reply:
It feels adequately documented on https://valkey.io/docs/topics/transactions/ in:
```
The server will detect an error during the accumulation of commands. It will then refuse to execute the transaction returning an error during EXEC, discarding the transaction.
```